docs/guides/connecting_to_your_data/cloud/s3/components_spark/_test_your_new_datasource.md

```diff
@@ -0,0 +1,46 @@
+import TabItem from '@theme/TabItem';
+import Tabs from '@theme/Tabs';
+import TechnicalTag from '@site/docs/term_tags/_tag.mdx';
+
+Verify your new <TechnicalTag tag="datasource" text="Datasource" /> by loading data from it into a <TechnicalTag tag="validator" text="Validator" /> using a <TechnicalTag tag="batch_request" text="Batch Request" />.
+
+<Tabs
+  defaultValue='runtime_batch_request'
+  values={[
+  {label: 'Specify an S3 path to single CSV', value:'runtime_batch_request'},
+  {label: 'Specify a data_asset_name', value:'batch_request'},
+  ]}>
+
+<TabItem value="runtime_batch_request">
+
+Add the S3 path to your CSV in the `path` key under `runtime_parameters` in your `RuntimeBatchRequest`.
+
+:::tip
+The path you will want to use is your S3 URI, not the URL.
+:::
+
+```python file=../../../../../../tests/integration/docusaurus/connecting_to_your_data/cloud/s3/spark/inferred_and_runtime_yaml_example.py#L57-L63
+```
+
+Then load data into the `Validator`.
+
+```python file=../../../../../../tests/integration/docusaurus/connecting_to_your_data/cloud/s3/spark/inferred_and_runtime_yaml_example.py#L71-L77
+```
+
+</TabItem>
+
+<TabItem value="batch_request">
+
+Add the name of the <TechnicalTag tag="data_asset" text="Data Asset" /> to the `data_asset_name` in your `BatchRequest`.
+
+```python file=../../../../../../tests/integration/docusaurus/connecting_to_your_data/cloud/s3/spark/inferred_and_runtime_yaml_example.py#L83-L88
+```
+
+Then load data into the `Validator`.
+
+```python file=../../../../../../tests/integration/docusaurus/connecting_to_your_data/cloud/s3/spark/inferred_and_runtime_yaml_example.py#L96-L102
+```
```

Comment:
It would be wonderful to get this as a named snippet instead of having the line numbers. Could it be within the scope of this PR?

Reply:
Connecting your Spark environment to S3 may require some additional configuration. 

First, we must ensure the following JAR dependencies are installed in the `Spark/jars` directory
  - [`hadoop-aws` jar that matches your Spark version](mvnrepository.com/artifact/org.apache.hadoop/hadoop-aws)
  - [`aws-java-sdk-bundle` jar that matches your Spark version](mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-bundle)

In addition, the following steps may be needed to update the Spark configuration: 

```python
conf = pyspark.SparkConf()
conf.set('spark.jars.packages', 'org.apache.hadoop:hadoop-aws:3.3.1')
```

And AWS credentials to the `SparkContext.`
```python
sc = SparkContext(conf=conf)
# sc._jsc.hadoopConfiguration().set('fs.s3a.aws.credentials.provider', 'org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider')
sc._jsc.hadoopConfiguration().set('fs.s3a.access.key', [AWS ACCESS KEY])
sc._jsc.hadoopConfiguration().set('fs.s3a.secret.key', [AWS SECRET KEY])
```

Additional information can be found in the [official documentation for Hadoop](https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-aws/src/site/markdown/tools/hadoop-aws/troubleshooting_s3a.md) and the following [StackOverflow discussion](https://stackoverflow.com/questions/58415928/spark-s3-error-java-lang-classnotfoundexception-class-org-apache-hadoop-f).

Reply:
I've added the extra step for getting those dependencies installed.

Unfortunately, updating the snippet would shift line numbers causing updates to be needed in a few other documents as well.  I think that falls out of scope, and should be a PR of its own.